pkg/vere/io/http.c

```diff
@@ -546,6 +553,57 @@ _http_req_dispatch(u3_hreq* req_u, u3_noun req)
   }
 }
 
+
+/* _http_cache_scry_cb(): insert scry result into noun cache
+*/
+static void
+_http_cache_scry_cb(void* vod_p, u3_noun nun)
+{
+  u3_httd* htd_u = vod_p;
+
+  // XX need both the scry path and the noun
+  u3_noun pax = XX;
+
+  u3h_put(htd_u->nax_p, pax, u3k(nun));
+  u3z(nun); // XX should the k and z cancel out?
+}
+
+
+/* _http_req_cache(): attempt to serve http request from cache
+*/
+static c3_o
+_http_req_cache(u3_hreq* req_u)
+{
+  c3_assert(u3_rsat_init == req_u->sat_e);
+  req_u->sat_e = u3_rsat_plan;
+
+  u3_httd* htd_u = req_u->hon_u->htp_u->htd_u;
+
+  u3_noun url = _http_vec_to_atom(req_u->rec_u->path);
+  u3_weak sac = u3h_get(htd_u->sax_p, url);
+  if ( u3_none == sac ) {
+    return c3n;
+  }
+
+  u3_weak nac = u3h_get(htd_u->nax_p, sac);
+  if ( u3_none == nac ) {
+    // noun not in cache; scry it
+    u3_pier_peek_last(htd_u->car_u.pir_u, u3_nul, c3__ex,
+                      u3_nul, sac, htd_u, _http_cache_scry_cb);
+    return c3n; // XX return c3y, and respond async in _http_cache_scry_cb?
+  }
+
+  // XX check auth
```

Comment:
IIUC, assuming https://github.com/urbit/vere/pull/199 is merged with its current API, the check should look like:
```c
if ( (c3y == auth) && (c3y == _http_req_is_auth(htd_u->fig_u, req_u->rec_u) ) {
   return c3n;
}
```